frontend/src/components/CommentSection.js

```diff
@@ -1,114 +1,89 @@
+            .then((data) => {
+                setComments([...data.data])
+                forceRefresh()
+                console.log(data)
+            })
+            .then(() => {
+                setLoading(false)
+            })
+            .catch(() => {
+                setLoading(false)
+            })
+    }
 
-const CommentSection = () => {
-    const renderElement = useCallback((props) => <Element {...props} />, [])
-    const renderLeaf = useCallback((props) => <Leaf {...props} />, [])
-    const editor = useMemo(() => withHistory(withReact(createEditor())), [])
+    useEffect(() => {
+        getComments(ticket)
+    }, [ticket])
 
     return (
-        <Box mt="20px">
-            <Heading textAlign="center">Comment Section </Heading>
-            <Slate editor={editor} initialValue={initialValue}>
-                <Toolbar>
-                    <MarkButton format="bold" icon="format_bold" />
-                    <MarkButton format="italic" icon="format_italic" />
-                    <MarkButton format="underline" icon="format_underlined" />
-                    <MarkButton format="code" icon="code" />
-                    <BlockButton format="heading-one" icon="looks_one" />
-                    <BlockButton format="heading-two" icon="looks_two" />
-                    <BlockButton format="block-quote" icon="format_quote" />
-                    <BlockButton
-                        format="numbered-list"
-                        icon="format_list_numbered"
-                    />
-                    <BlockButton
-                        format="bulleted-list"
-                        icon="format_list_bulleted"
-                    />
-                    <BlockButton format="left" icon="format_align_left" />
-                    <BlockButton format="center" icon="format_align_center" />
-                    <BlockButton format="right" icon="format_align_right" />
-                    <BlockButton format="justify" icon="format_align_justify" />
-                </Toolbar>
-                <Editable
-                    renderElement={renderElement}
-                    renderLeaf={renderLeaf}
-                    placeholder="Enter some rich text…"
-                    spellCheck
-                    autoFocus={false}
-                    onKeyDown={(event) => {
-                        for (const hotkey in HOTKEYS) {
-                            if (isHotkey(hotkey, event)) {
-                                event.preventDefault()
-                                const mark = HOTKEYS[hotkey]
-                                toggleMark(editor, mark)
-                            }
-                        }
-                    }}
+        <Box mt="50px" mb="50px">
+            <Box
+                fontWeight="bold"
+                fontSize="30px"
+                textAlign="left"
+                marginLeft="5px"
+            >
+                Comments
+            </Box>
+            <div key={refreshKey}>
```

Comment:
what is the purpose of this `refreshKey`?

Reply:
Slate editor doesn't rerender the actual comment values once getComments() is called again when mapping through comments

{comments.map((content, index) => {
    return (
        <CommentView
            key={
                'comment-' +
                (comments.length - index).toString() +
                ticket
            }
            comment={content}
            allUsers={allUsers}
            getComments={getComments}
        />
    )
})}

Reply:
So I opted to force rerender to change the actual values